### DISCUSSION TITLE
Decode Quality and Check Box Questions

### DISCUSSION THREAD

**Discussion by gcowie85** [latenitefilms/BRAWToolbox#76] [Support] [answered]
Decode Quality is a bit confusing to me and I'm sure others. In the drop down the order is
-Automatic (HD)
-Autumatic (Ultra HD)
-Full

Isn't Ultra the best out of those 3? I personally don't know but from what I read online that's what I gathered. If that's so, I would suggest putting them in that order since I think that order would make sense to people who don't understand. Maybe a little tab next to the top one saying (Highest) and then (Lowest) next to the lowest quality to help guide people. 

**Check Boxes**
If a box is not checked does that mean that change won't take place? Like if I change that original Decode Quality from HD to Full but don't have that box check it won't take place when I bring it into FCPX?

Also I noticed  you have to select each individual box for each clip before you paste. This can be time consuming, could we not just copy paste "all" if we choose to all clips we are importing and then make minor tweaks from there?

Amazing product by the way! Wrote you a review raving about how we needed this for years. Sad Apple or BM never did this. Thanks for stepping up and being a genius!

**Comment by latenitefilms** [comment, marked as answer]
This is all explained here: https://brawtoolbox.io/workflow-extension/#inspector

The automatic decode quality basically uses the Blackmagic SDK to pick the best decode quality for the destination resolution (i.e. HD or UltraHD).

For example, if you have it set to **Automatic (HD)**, if you have a HD BRAW clip, it'll use Full Quality, however if you have a 12K BRAW file, it'll use a lesser decode quality, because the source pixels are so large.

Generally speaking, I normally just leave it on **Automatic (HD)** for editing - as I don't need anything better than HD quality, then I force it to **Full** for final exports.

The checkboxes next to each parameter are selection options (as opposed to something that might enable/disable something).

You can use the **Apply Checked to All** if you want to change the properties of every BRAW clip in your file list.

**Reply by leightv** [reply]
Hi Chris,
I'm a little confused about this as well. So if it's set to Automatic (HD) you're seeing an HD version of your clip when editing. How do you force it to Full when you're doing your final exports?

**Reply by latenitefilms** [reply]
Apologies for the very delayed reply @leightv - I totally missed this message. Sorry!

> So if it's set to Automatic (HD) you're seeing an HD version of your clip when editing.

Not exactly. The resolution never changes inside Final Cut Pro. If you have a 6K BRAW file, Final Cut Pro will always treat it as 6K. However, the decode quality affects the quality of the debayer/rendering process. Using **Automatic (HD)** basically just tries to pick the best decode quality for HD delivery. For example, if you're working with 12K BRAW files, and you have **Automatic (HD)** picked, it'll use something less than Full Quality, because the footage is being downsized to HD anyway. The actual maths/algorithm is all handled within Blackmagic's closed-source SDK, so I can't actually tell you exactly what it would pick - and it may change between camera models and SDK versions, etc. We basically just trust that Blackmagic is making the right choices. 

> How do you force it to Full when you're doing your final exports?

You can click the **Global Settings** button within the Final Cut Pro Inspector on ANY of your BRAW clips to "force" a decode quality. Clicking one of these "force" buttons basically forces the BRAW Toolbox render to ALWAYS use this decode quality across all clips.
